### PR TITLE
Codecov migration

### PR DESCRIPTION
**Pull request by alanmcruickshank**
This might fail but I hope it works.

The reason codecov isn't reporting is that the `Python 3.10 Windows tests` job isn't reporting. The error we're getting is `{'detail': ErrorDetail(string='Missing "owner" argument. Please upload with the Codecov repository upload token to resolve issue.', code='not_found')}`. We're currently using the v1 uploader and that has been officially sunset: https://github.com/codecov/codecov-action.

This updates to v3 (and hopefully solves our codecov issues).

See here for example: https://github.com/sqlfluff/sqlfluff/actions/runs/3749312506/jobs/6367654414

### CODE REVIEW

**Comment by alanmcruickshank** [comment]
Up to date thread on codecov: https://community.codecov.com/t/upload-issues-unable-to-locate-build-via-github-actions-api/3954

Suggests adding a token to resolve. So that this works for PRs from forks, I'm going to add a token which is public. I think the security risks of doing so are minimal. FerretDB have done exactly that: https://github.com/FerretDB/FerretDB/blob/f4de6e41fc6d0ab9f00ac9e2facdda877fca86ee/.github/workflows/go.yml#L23-L25

File: src/sqlfluff/core/default_config.cfg [hunk 1]
```diff
@@ -63,6 +63,7 @@ indented_joins = False
 indented_ctes = False
 indented_using_on = True
 indented_on_contents = True
+allow_implicit_indents = False
```
Comment (barrywhart) [inline src/sqlfluff/core/default_config.cfg:66]: Did you mean to include this line? Seems unrelated to coverage checks.

**Review by barrywhart — APPROVED** [review]

File: src/sqlfluff/core/default_config.cfg (hunk 1 above)
Comment (alanmcruickshank) [reply src/sqlfluff/core/default_config.cfg:66]: oops - no I didn't but it's not harmful 🤷
